internal/commands/vaultsecrets/run/run.go

```diff
@@ -0,0 +1,157 @@
+		PreviewClient: preview_secret_service.New(ctx.HCP, nil),
+		Client:        secret_service.New(ctx.HCP, nil),
+	}
+
+	cmd := &cmd.Command{
+		Name:      "run",
+		ShortHelp: "Run a process with secrets from a Vault Secrets app.",
+		LongHelp: heredoc.New(ctx.IO).Must(`
+		The {{ template "mdCodeOrBold" "hcp vault-secrets run" }} command lets you
+		run the provided command as a child process while injecting
+        all of the app's secrets as environment variables, with all secret names
+        converted to upper-case. The stdout/stderr from the child
+        process are forwarded to the top level {{ template "mdCodeOrBold" "hcp vault-secrets run" }} command.
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: `Inject secrets as an environment variable:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets run "env"
+				`),
+			},
+		},
+		Args: cmd.PositionalArguments{
+			Args: []cmd.PositionalArgument{
+				{
+					Name:          "COMMAND",
+					Documentation: "Defines the invocation of the child process to inject secrets to.",
+				},
+			},
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "app",
+					DisplayValue: "NAME",
+					Description:  "The application you want to pull all secrets from.",
+					Value:        appname.Flag(),
+				},
+			},
+		},
+		PersistentPreRun: func(c *cmd.Command, args []string) error {
+			return appname.Require(ctx)
+		},
+		RunF: func(c *cmd.Command, args []string) error {
+			opts.Command = args[0]
+
+			opts.AppName = appname.Get()
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runRun(opts)
+		},
+	}
+	cmd.Args.Autocomplete = helper.PredictAppName(ctx, cmd, preview_secret_service.New(ctx.HCP, nil))
+
+	return cmd
+}
+
+func runRun(opts *RunOpts) (err error) {
+	envSecrets, err := getAllSecretsForEnv(opts)
+	if err != nil {
+		return fmt.Errorf("failed to run with secrets in app %q: %w", opts.AppName, err)
+	}
+
+	childProcess := setupChildProcess(opts.Ctx, opts.Command, envSecrets)
+	err = childProcess.Run()
+	if err != nil {
+		return fmt.Errorf("failed to run with secrets in app %q: %w", opts.AppName, err)
+	}
+
+	return nil
+}
+
+func getAllSecretsForEnv(opts *RunOpts) ([]string, error) {
+	params := preview_secret_service.NewOpenAppSecretsParamsWithContext(opts.Ctx)
+	params.OrganizationID = opts.Profile.OrganizationID
+	params.ProjectID = opts.Profile.ProjectID
+	params.AppName = opts.AppName
+
+	res, err := opts.PreviewClient.OpenAppSecrets(params, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// get existing environment variables detectable by runtime
+	result := os.Environ()
+
+	for _, secret := range res.Payload.Secrets {
+		// we need to append results in case of duplicates we want secrets to override
+		// only supporting static secrets
+		if secret.StaticVersion != nil {
+			result = append(result, fmt.Sprintf("%v=%v", strings.ToUpper(secret.Name), secret.StaticVersion.Value))
+
+		}
+	}
```

Comment:
if you're using the preview api (2023-11-23 version) `OpenAppSecrets` will return rotating secrets as well
can see how we handled that in `run` here
https://github.com/hashicorp/vlt/blob/b9ce887fec166763cb715bca8fe9be46ee24b3f1/internal/command/run.go#L172-L183